### ISSUE TITLE
How to add list of recent posts / archives in a new page?

### ISSUE THREAD

**Issue by dennis-leeyinghui** [hexojs/hexo#4681]
## Check List

Please check followings before submitting a new issue.

- [V] I have already read [Docs page](https://hexo.io/docs/) & [Troubleshooting page](https://hexo.io/docs/troubleshooting)
- [V] I have already searched existing issues and they are not help to me
- [V] I examined error or warning messages and it's difficult to solve
- [V] Using [the latest](https://github.com/hexojs/hexo/releases) version of Hexo (run `hexo version` to check)
- [V] Node.js is higher than [minimum required version](https://hexo.io/docs/#Minimum-required-Node-js-version)

## Question

I have read through tutorials and different questions on Github / Stackoverflow, but I'm still quite puzzled how I can add the list of posts ( either recent posts, or archives) to a new page?

In other words, how can I designate a page to use certain widgets?

Thanks.

**Comment by dennis-leeyinghui** [comment]
I've tried to change the layout to "archive", but it doesn't work.

---
**layout: archive
title: my title
date: 2021-04-05 15:04:02**
---

The error in CMD reads:

```
....theme\mango\layout\_partial\archive.ejs:7
    5| <% } else { %>
    6|   <% var last; %>
 >> 7|   <% page.posts.each(function(post, i){ %>
    8|     <% var year = post.date.year(); %>
    9|     <% if (last != year){ %>
    10|       <% if (last != null){ %>

Cannot read property 'each' of undefined at ... themes\mango\layout\_partial\archive.ejs:28:19)
```

**Comment by dennis-leeyinghui** [comment]
It seems I've found a workaround for now.

There are two things I have done:

**(1) Change my Homepage**

Go to root path\source\index.md (create one if you don't have it)

It will become your homepage.

**(2) Modify your root path\_config.yml**

Look for the line: 

```
index_generator:
  path: ''
```
 
Change the path to your designated page to populate the post list.
